### ISSUE TITLE
[a11y]: Selectable Tile with AI label causes invalid markup

### ISSUE THREAD

**Issue by janhassel** [carbon-design-system/carbon#17237]
### Package

@carbon/react

### Package version

1.61.0

### Automated testing tool and ruleset

Manual testing

### Description

When adding an AI label to a `SelectableTile`, the AI label and therefore a popover triggered by a `button` are descendants of a `role="checkbox"`:

```html
<div
  class="cds--tile cds--tile--selectable cds--tile--slug slug-selectable-tile"
  role="checkbox"
  …
>
  <span><svg>…</svg></span>

  <label>…</label>

  <div class="cds--slug">
    <span class="cds--popover-container">
      <button>…</button>
      <span class="cds--popover">…</span>
     </span>
  </div>
</div>
```

According to [WAI-ARIA 1.2](https://www.w3.org/TR/wai-aria-1.2/#checkbox), all descendants of a `role="checkbox"` are presentational, meaning the button loses its semantics.

[ARIA in HTML](https://www.w3.org/TR/html-aria/#index-aria-checkbox) also specifically states the allowed descendants elements with a checkbox role:
> Phrasing content, but with no interactive content descendants, and no descendants with a tabindex attribute specified.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-tile--with-ai-label

### Steps to reproduce

See markup pasted above

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Thanks for reporting this one, it's the same thing we're running into with the table header cells https://github.com/carbon-design-system/carbon/issues/16031 

Might be that the only way to avoid this is to move the AILabel outside the tile dom and then visually reposition it atop the tile.
